File: Submission-Java/app/src/main/java/com/ahmadabuhasan/appgithubuser/UserDetailActivity.java
package com.ahmadabuhasan.appgithubuser;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.util.TypedValue;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import com.ahmadabuhasan.appgithubuser.databinding.ActivityUserDetailBinding;
import com.bumptech.glide.Glide;

import java.util.Objects;

public class UserDetailActivity extends AppCompatActivity {

    public static final String EXTRA_USER = "extra_user";
    User user = new User();

    ActivityUserDetailBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_user_detail);
        setTitle("Detail User");

        Objects.requireNonNull(getSupportActionBar()).setHomeButtonEnabled(true);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        binding = ActivityUserDetailBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        user = getIntent().getParcelableExtra(EXTRA_USER);
        int imgAvatar = user.getAvatar();
        String textName = user.getName();
        String textUsername = user.getUsername();
        String textLocation = user.getLocation();
        String textRepository = user.getRepository();
        String textCompany = user.getCompany();
        String textFollowers = user.getFollowers();
        String textFollowing = user.getFollowing();

        Glide.with(this)
                .load(imgAvatar)
                .into(binding.imgAvatarReceived);
        binding.tvNameReceived.setText(textName);
        binding.tvUsernameReceived.setText(String.format("@%s", textUsername));
        binding.tvLocationReceived.setText(String.format("Location : %s", textLocation));
        binding.tvCompanyReceived.setText(String.format("Company : %s", textCompany));

        binding.repositoryReceived.setTextColor(Color.BLACK);
        binding.repositoryReceived.setTextSize(TypedValue.COMPLEX_UNIT_SP, 10);
        binding.repositoryReceived.setText(String.format("Repository \n \n%s", textRepository));

        binding.followersReceived.setTextColor(Color.BLACK);
        binding.followersReceived.setTextSize(TypedValue.COMPLEX_UNIT_SP, 10);
        binding.followersReceived.setText(String.format("Followers \n \n%s", textFollowers));

        binding.followingReceived.setTextColor(Color.BLACK);
        binding.followingReceived.setTextSize(TypedValue.COMPLEX_UNIT_SP, 10);
        binding.followingReceived.setText(String.format("Following \n \n%s", textFollowing));
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.optionmenu, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(@NonNull MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            onBackPressed();
        } else if (item.getItemId() == R.id.share) {
            String shareIntent = "Github User" +
                    "\n Name: " + user.getName() +
                    "\n Username: " + user.getUsername() +
                    "\n Company: " + user.getCompany() +
                    "\n Location: " + user.getLocation() +
                    "\n Repository: " + user.getRepository() +
                    "\n Followers: " + user.getFollowers() +
                    "\n Following: " + user.getFollowing();
            Intent i = new Intent();
            i.setAction(Intent.ACTION_SEND);
            i.putExtra(Intent.EXTRA_TEXT, shareIntent);
            i.setType("text/plain");
            startActivity(Intent.createChooser(i, "Share using"));
        }
        return true;
    }
}

File: Submission-Kotlin/app/src/main/java/com/ahmadabuhasan/appgithubuser/ui/UserDetailActivity.kt
package com.ahmadabuhasan.appgithubuser.ui

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.ahmadabuhasan.appgithubuser.R
import com.ahmadabuhasan.appgithubuser.databinding.ActivityUserDetailBinding
import com.ahmadabuhasan.appgithubuser.model.UserResponse
import com.bumptech.glide.Glide

class UserDetailActivity : AppCompatActivity() {

    companion object {
        const val EXTRA_USER = "extra_user"
    }

    private lateinit var binding: ActivityUserDetailBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityUserDetailBinding.inflate(layoutInflater)
        setContentView(binding.root)

        supportActionBar?.title = "Detail User"
        supportActionBar?.setHomeButtonEnabled(true)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)
        supportActionBar?.setHomeAsUpIndicator(R.drawable.ic_arrow_back)

        val user = intent.getParcelableExtra<UserResponse>(EXTRA_USER)
        if (user != null) {
            Glide.with(this)
                .load(user.avatar)
                .skipMemoryCache(true)
                .into(binding.imgAvatar)

            binding.tvName.text = user.name
            binding.tvUsername.text = buildString {
                append("@ ")
                append(user.username)
            }
            binding.tvCompany.text = user.company
            binding.tvLocation.text = user.location
            binding.tvRepositoryValue.text = user.repository.toString()
            binding.tvFollowersValue.text = user.follower.toString()
            binding.tvFollowingValue.text = user.following.toString()
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}